webapp/tests/karma/ts/services/rules-engine.service.spec.ts

```diff
@@ -475,7 +480,7 @@ describe('RulesEngineService', () => {
 
     expect(rulesEngineCoreStubs.fetchTasksFor.callCount).to.eq(1);
     expect(rulesEngineCoreStubs.fetchTargets.callCount).to.eq(1);
-    expect(telemetryService.record.callCount).to.equal(5);
+    expect(telemetryService.record.callCount).to.equal(6);
```

Comment:
Hmm, how come this changed? What's been added?

Reply:
Above in the configuration of the tests there was this ([here](https://github.com/medic/cht-core/pull/6771/files/b4db569b9a336446a42dccfbf1517f70fb64ce04#diff-8c10291e83893a44e49518280fe5dcff0b1b55d707a430e25465dd5d1ea2d0bbR101) and [here](https://github.com/medic/cht-core/pull/6771/files/b4db569b9a336446a42dccfbf1517f70fb64ce04#diff-8c10291e83893a44e49518280fe5dcff0b1b55d707a430e25465dd5d1ea2d0bbR114)):

```js
    fetchTasksResult = Promise.resolve;
```

That I changed for this:

```js
    fetchTasksResult = () => Promise.resolve();
```

Both look like the same but the first case was causing the loop executed by the test cases to fail silently (actually was logging out errors but not causing the test cases to fail).